modules/apps/27-interchain-accounts/controller/keeper/handshake_test.go

```diff
@@ -471,3 +477,192 @@ func (suite *KeeperTestSuite) TestOnChanCloseConfirm() {
+			metadata.Encoding = icatypes.EncodingProto3JSON
+
+			tc.malleate() // malleate mutates test data
+
+			version := string(icatypes.ModuleCdc.MustMarshalJSON(&metadata))
+
+			upgradeVersion, err := path.EndpointA.Chain.GetSimApp().ICAControllerKeeper.OnChanUpgradeInit(
+				path.EndpointA.Chain.GetContext(),
+				path.EndpointA.ChannelConfig.PortID,
+				path.EndpointA.ChannelID,
+				[]string{path.EndpointA.ConnectionID},
+				version,
+			)
+
+			expPass := tc.expError == nil
+
+			if expPass {
+				suite.Require().NoError(err)
+				suite.Require().Equal(upgradeVersion, version)
+			} else {
+				suite.Require().ErrorIs(err, tc.expError)
+			}
+		})
+	}
+}
+
+func (suite *KeeperTestSuite) TestOnChanUpgradeTry() {
+	var (
+		path     *ibctesting.Path
+		metadata icatypes.Metadata
+	)
+
+	testCases := []struct {
+		name     string
+		malleate func()
+		expError error
+	}{
+		{
+			"success",
+			func() {},
+			nil,
+		},
+		{
+			name: "failure: invalid port ID",
+			malleate: func() {
+				path.EndpointB.ChannelConfig.PortID = "invalid-port-id"
+			},
+			expError: porttypes.ErrInvalidPort,
+		},
+		{
+			name: "failure: empty counterparty version",
+			malleate: func() {
+				channel := path.EndpointA.GetChannel()
+				channel.Version = ""
+				path.EndpointA.SetChannel(channel)
+			},
+			expError: channeltypes.ErrInvalidChannelVersion,
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc
+
+		suite.Run(tc.name, func() {
+			suite.SetupTest() // reset
+
+			path = NewICAPath(suite.chainA, suite.chainB)
+			suite.coordinator.SetupConnections(path)
+
+			err := SetupICAPath(path, TestOwnerAddress)
+			suite.Require().NoError(err)
+
+			channel := path.EndpointA.GetChannel()
+
+			currentMetadata, err := icatypes.ParseMedataFromString(channel.Version)
+			suite.Require().NoError(err)
+
+			metadata = icatypes.NewDefaultMetadata(path.EndpointA.ConnectionID, path.EndpointB.ConnectionID)
+			// use the same address as the previous metadata.
+			metadata.Address = currentMetadata.Address
+
+			// this is the actual change to the version.
+			metadata.Encoding = icatypes.EncodingProto3JSON
+
+			path.EndpointA.ChannelConfig.ProposedUpgrade.Fields.Version = string(icatypes.ModuleCdc.MustMarshalJSON(&metadata))
+			path.EndpointB.ChannelConfig.ProposedUpgrade.Fields.Version = string(icatypes.ModuleCdc.MustMarshalJSON(&metadata))
+
+			err = path.EndpointA.ChanUpgradeInit()
+			suite.Require().NoError(err)
+
+			tc.malleate() // malleate mutates test data
+
+			version, err := suite.chainB.GetSimApp().ICAHostKeeper.OnChanUpgradeTry(
+				suite.chainB.GetContext(),
+				path.EndpointB.ChannelConfig.PortID,
+				path.EndpointB.ChannelID,
+				channeltypes.ORDERED,
+				[]string{path.EndpointB.ConnectionID},
+				path.EndpointA.GetChannel().Version,
+			)
+
+			expPass := tc.expError == nil
+			if expPass {
+				suite.Require().Equal(path.EndpointA.GetChannel().Version, version)
+				suite.Require().NoError(err)
```

Comment:
Maybe the other way around?

```suggestion
                                suite.Require().NoError(err)
				suite.Require().Equal(path.EndpointA.GetChannel().Version, version)
```